src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs

```diff
@@ -318,6 +318,8 @@ public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken
                 {
                     if (_readLen == _readPos && buffer.Length >= _bufferSize)
                     {
+                        // invalidate the buffered data, otherwise certain Seek operation followed by a ReadAsync could try to re-use data from _buffer
+                        _readPos = _readLen = 0;
```

Comment:
This is on the ReadAsync path. Is there any similar issue on the Read path? What about in BufferedStream? Just want to make sure we've audited all similar code paths that could have a similar issue.

Reply:
> Is there any similar issue on the Read path?

There is not, and the test that I've added ensures this.

> What about in BufferedStream?

This particular optimization was not ported to `BufferedStream`. But I can add a `BufferedStream` test to be 100% sure.

> Just want to make sure we've audited all similar code paths that could have a similar issue.

Sure, that is a very good point. I've verified that neither Read or Write/WriteAsync suffer from same issue.